### ISSUE TITLE
BUILD SYSTEM: Free CCI accounts can't run integration tests

### ISSUE THREAD

**Issue by tlimoncelli** [StackExchange/dnscontrol#1935]
Our integration tests use "xlarge" instances.  However that size is not available in CircleCI "free tier" accounts.

We can't change to "large" as a default because built times are 2x or 3x longer on the smaller instances.

The workaround is that free tier users maintain a separate config.yml file that changes 2xlarge to large.  It is inconvenient to make this change and prevent it from getting merged into the main repo.
 
It would be nice if free tier users could magically use large instances instead, or at least have a flag or env they can set to change it.

Followup: Whatever the solution is, please document it in docs/release-engineering.md

**Comment by tlimoncelli** [comment]
CC @cdhunt

**Comment by cdhunt** [comment]
This should be covered by our corporate agreement. Maybe it doesn't cover forks. Are you seeing this in the StackExchange org? I'll have a look.

**Comment by tlimoncelli** [comment]
No, this is with my personal accounts.

I tried to emulate a situation where a community member (someone who is not a stack employee) wants to "BYO creds".  I used my personal github account (tomontime) to fork the project. I then created a free account on CircleCI.  Everything was going fine until a 2xlarge instance was needed ([link](https://app.circleci.com/pipelines/github/TomOnTime/dnscontrol/2/workflows/0b3341c3-23ef-43ae-8785-ebe3dec7354b)).  In my fork I changed the instance size to `large` and things worked ([link](https://app.circleci.com/pipelines/github/TomOnTime/dnscontrol/3/workflows/aadcad3a-99b6-403b-9096-3f21489d385d))

I don't think we should extend our corporate agreements to forks outside of the StackEnterprise org.  Someone could fork the project and change the code to something we don't want to pay for.
